### PR TITLE
[Refactor:System] Separate UTM variable from ARM

### PR DESCRIPTION
**Pull request by skara9**
#### Current Behavior
The VM installation scripts currently check for an `aarch64` architecture and assign the shell variable `UTM_ARM` to track it.
Also, the install script skips installing haskell stack for ARM machines.

#### Updated Behavior
* A separate `--utm` flag has been added to indicate running on UTM, differentiating between ARM configurations not running on UTM and configurations of any architecture running on UTM.
* A `DEV_VM` variable has been created to indicate a script running on either Vagrant or UTM.
* Furthermore, the install script has been updated to install haskell stack for ARM given that stack ships an ARM binary since [v2.9.1](https://github.com/commercialhaskell/stack/releases/tag/v2.9.1).

Partly addresses #7885

### CODE REVIEW

**Comment by bmcutler** [comment]
@MasterOdin Can you help review this please? thanks

File: .setup/install_system.sh [hunk 1]
```diff
@@ -59,6 +59,7 @@ source ${CURRENT_DIR}/bin/versions.sh
 # PROVISION SETUP
 #################
 
+export UTM=0
```
Comment (MasterOdin) [inline .setup/install_system.sh:62]: add `export DEV_VM=0` up here as well.

**Review by MasterOdin — CHANGES_REQUESTED** [review]
Changes overall seem reasonable given the concept that we are potentially supporting multiple different VM platforms. Some minor comments.

**Comment by tkoz0** [comment]
I tested `vagrant up` on this branch and it works as expected. My host OS is Ubuntu 22.04 on an Intel x86_64 CPU and my hypervisor for the VM is VirtualBox 6.1.

**Comment by bmcutler** [comment]
@MasterOdin could you please re-review?  Thanks!

**Review by EdenBendheim — COMMENTED** [review]
I ran into an installation error:
image

Not too sure of how to fix it. Looked around in Install_system and the files for installing it were there.

**Comment by MasterOdin** [comment]
@EdenBendheim You're hitting that error when going through UTM? I think we stopped trying to install that package from Ubuntu 20.04 and onwards though so I'm not sure why you're hitting that assuming you're installing Ubuntu 20.04. If you're trying to install an older Ubuntu, that's unsupported.

**Pull request merged**